### ISSUE TITLE
[BUG] Module only looks for default value in SchemaType.options.defaultValue

### ISSUE THREAD

**Issue by makinde** [DouglasGabr/mongoose-lean-defaults#3]
When I can `mySchemaType.default(null)`, the default value gets stored in `mySchemaType.default`

**Comment by DouglasGabr** [comment]
@makinde thanks for the bug report, I'll take a look at it as soon as I can. Feel free to open a PR if you want to

**Comment by makinde** [comment]
I'm using mongoose 5.4.0, btw.

Thanks for the quick reply. I might be able to get to it in the next two weeks, but not sure I'd be able to flesh out unit tests and such.

**Comment by makinde** [comment]
I'm patching in my code for now

```js
  // Mark all the the non-required fields as defaulting to null/empty object
  schema.eachPath((pathName, schemaType) => {
    const type = schema.pathType(pathName);
    if (!schemaType.isRequired && schemaType.defaultValue === undefined) {
      const defaultVal = (type === 'nested') ? {} : null;
      schemaType.default(defaultVal);

      // Store the default directly in options so the leanDefaults works properly
      // Until this is fixed: https://github.com/DouglasGabr/mongoose-lean-defaults/issues/3
      if (!schemaType.options) schemaType.options = {};
      schemaType.options.default = defaultVal;
    }
  });
```

**Comment by DouglasGabr** [comment]
Alright, thanks for sharing your code, I'll publish a new release today or tomorrow

**Comment by DouglasGabr** [comment]
Fixed in `v0.3.1`. Please reopen the issue if something is still not working

**Comment by makinde** [comment]
Awesome, thanks for the fix!
